src/helperFunctions/hasSubArray.ts

```diff
@@ -0,0 +1,11 @@
+export const hasSubArray = (
```

Comment:
I think `hasSubArray` would be better off as something like `includesSubArray` -- I honestly can't parse *why* but `hasSubArray` to me sounds like something that would *always* be true unless the array is empty (eg, every array containing arrays has a sub array of some kind), whereas `includes` mirrors the `.includes` method already on the `Array` prototype